### PR TITLE
fix(suite-native): searchable accounts

### LINKED ISSUE #6988: Filter imported accounts
As a user I would like to be able to filter / search in imported assets in order to be able to find my assets faster when I have dozens of them. 

Figma: https://www.figma.com/file/Z6AGVUmKQzLNtDozFamW7f/%5Bs2.0%5D-Mobile?type=design&node-id=19433%3A112322&mode=design&t=oX0r9J8ejKIZrzpt-1

Acceptance criteria: 

- User can search via name of the account, name of the coin
- Matching results are displayed

Add filter for imported accounts based on stored xpubs. If user adds new type of coin, it should be included as filtering option in `My assets` tab.

### PR DESCRIPTION
**Pull request by PeKne**
Follow-up of PR #9098

- UI of the search bar was adjusted to match the design.
- Filtering of included tokens matches only if the token has a fiat rate value.
- Filtering was added also to the Receive screen.

Fixes #6988

## Screenshots:

https://github.com/trezor/trezor-suite/assets/26143964/c3856282-3c2d-4576-8b2f-1aafda83eefe

### CODE REVIEW

File: suite-native/atoms/src/Input/SearchInput.tsx [hunk 1]
```diff
@@ -54,7 +46,13 @@ const inputWrapperStyle = prepareNativeStyle<InputStyleProps>((utils, { isFocuse
     ],
 }));
 
-export const SearchInput = ({ value, onChange, placeholder, isDisabled = false }: InputProps) => {
+export const SearchInput = ({
+    value,
+    onChange,
+    placeholder,
+    isDisabled = false,
```
Comment (juriczech) [inline suite-native/atoms/src/Input/SearchInput.tsx]: default values props should always be as last prop item

File: suite-native/accounts/src/selectors.ts [hunk 2]
```diff
@@ -32,18 +35,34 @@ export const isFilterValueMatchingAccount = (account: Account, filterValue: stri
 /**
  * Filter accounts by labels, network names and included token names.
  */
-const filterAccountsByLabelAndNetworkNames = (accounts: Account[], filterValue?: string | null) => {
+const filterAccountsByLabelAndNetworkNames = (
+    accounts: readonly Account[],
```
Comment (juriczech) [inline suite-native/accounts/src/selectors.ts:39]: Is the readonly really necessary? And why are we using it in some selectors and not others? I'd drop it as it's not really useful and you get a runtime error when trying to modify readonly state anyway

File: suite-native/accounts/src/selectors.ts [hunk 3]
```diff
@@ -32,18 +35,34 @@ export const isFilterValueMatchingAccount = (account: Account, filterValue: stri
...
+    filterValue?: string | null,
```
Comment (juriczech) [inline suite-native/accounts/src/selectors.ts]: Why is this `undefined | null | string`? In the AccountList you have the filter value defined as string.

File: suite-native/accounts/src/selectors.ts [hunk 4]
```diff
@@ -32,18 +35,34 @@ export const isFilterValueMatchingAccount = (account: Account, filterValue: stri
...
     if (!filterValue) return accounts;
 
     return A.filter(accounts, account => isFilterValueMatchingAccount(account, filterValue));
 };
 
 export const selectFilteredAccountsGroupedByNetwork = memoizeWithArgs(
-    (state: AccountsRootState, filterValue?: string | null) => {
+    (
+        state: AccountsRootState & FiatRatesRootState & SettingsSliceRootState,
+        filterValue?: string | null,
```
Comment (juriczech) [inline suite-native/accounts/src/selectors.ts]:
```suggestion
        filterValue: string,
```

File: suite-native/accounts/src/selectors.ts (hunk 3 above)
Comment (PeKne) [reply suite-native/accounts/src/selectors.ts]: Good point. Originally it was possibly undefined | null. Now after refactoring olny valid input is string. a14e10331

File: suite-native/accounts/src/selectors.ts (hunk 4 above)
Comment (PeKne) [reply suite-native/accounts/src/selectors.ts]: Same as above. Fixed a14e10331

File: suite-native/atoms/src/Input/SearchInput.tsx (hunk 1 above)
Comment (PeKne) [reply suite-native/atoms/src/Input/SearchInput.tsx]: Order changed in a14e10331.

File: suite-native/accounts/src/selectors.ts (hunk 2 above)
Comment (PeKne) [reply suite-native/accounts/src/selectors.ts:39]: I am afraid it has to be `readonly`. This is (IMO) a very annoying feature of `ts-belt` that everything that comes out of any of its functions is always set with `readonly` flag. If I set the `accounts` param without readonly, the TSlint does not like it.

Screenshot 2023-08-11 at 10 10 03

**Review by juriczech — APPROVED** [review]

**Review by juriczech — APPROVED** [review]

**Pull request merged**